watcher/lib/influxdb/influxdb.go

```diff
@@ -136,19 +141,86 @@ func (c *Client) UpdateRollup(r Rollup) error {
 	query := strings.Join([]string{deleteQuery, createQuery}, "; ")
 	log.Infof("%v", query)
 
-	if err = c.postQuery(query); err != nil {
+	if _, err = c.postQuery(query); err != nil {
 		return trace.Wrap(err)
 	}
 	return nil
 }
 
-func (c *Client) postQuery(query string) error {
+func (c *Client) postQuery(query string) (*client_v2.Response, error) {
 	response, err := c.client.Query(client_v2.NewQuery(query, "", ""))
 	if err != nil {
-		return trace.Wrap(err)
+		return nil, trace.Wrap(err)
 	}
 	if response.Error() != nil {
-		return trace.Wrap(response.Error())
+		return nil, trace.Wrap(response.Error())
+	}
+
+	return response, nil
+}
+
+func (c *Client) createDatabase() error {
+	showDatabasesResponse, err := c.postQuery(showDatabasesQuery)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	var createDatabase = true
+	// loop over response for `SHOW DATABASES`
+	for _, result := range showDatabasesResponse.Results {
+		for _, row := range result.Series {
+			if row.Values != nil && row.Values[0][0].(string) == constants.InfluxDBDatabase {
+				createDatabase = false
+				break
+			}
+		}
+	}
+
+	if createDatabase {
+		log.Infof("%v", fmt.Sprintf(createDatabaseQuery, constants.InfluxDBDatabase, constants.DurationDefault))
+		if _, err := c.postQuery(fmt.Sprintf(createDatabaseQuery, constants.InfluxDBDatabase, constants.DurationDefault)); err != nil {
+			return trace.Wrap(err)
+		}
+	}
+
+	return nil
+}
+
+func (c *Client) createRetentionPolicies() error {
```

Comment:
Also, as I read the [documentation](https://docs.influxdata.com/influxdb/v1.6/query_language/database_management/#create-retention-policies-with-create-retention-policy) it seemed that `influxdb` would only complain when creating a retention policy with the same name as existing **but** different attributes. In other two cases (i.e. when creating a new retention policy or creating a retention policy identical to the existing one under a new name **does not** return an error).
If the cause for `retention policy already exists` is a different set of attributes, skipping over the names will actually **not** create the retention policy as is intended by the `Setup` code or am I missing anything?